packages/angular_devkit/build_angular/src/differential-loading/is-differential-loading-needed.ts

```diff
@@ -0,0 +1,138 @@
+      + Array.from(visited).join(', ') );
+  }
+  visited.add(tsConfigFileName);
+
+  const content = require(tsConfigFileName);
+  const compilerOptions = content['compilerOptions'];
+  const target = compilerOptions['target'] as string;
+  const extendsField = content['extends'];
+
+  if (compilerOptions && target) {
+
+    const targetUC = target.toUpperCase();
+
+    // @ts-ignore
+    const scriptTarget = ScriptTarget[targetUC] as ScriptTarget;
+
+    if (typeof scriptTarget === 'undefined') {
+      throw new Error('Unsupported script target in tsconfig.json found: ' + target);
+    }
+
+    return scriptTarget;
+  } else if (extendsField) {
+    const tsConfigDirName = path.dirname(tsConfigFileName);
+    const nextTsConfigFileName = path.join(tsConfigDirName, extendsField);
+
+    return _getTargetVersion(nextTsConfigFileName, visited);
+  } else {
+    // Fallback to ES5
+    return ScriptTarget.ES5;
+  }
+
+}
+
+export function isEs5BrowserSupportNeeded(
+  tsConfigFileName: string | null, queries: string[] | undefined = undefined) {
+
+  const options: browserslist.Options = {};
+  if (tsConfigFileName) {
+    options.path = path.dirname(tsConfigFileName);
+  }
+
+  const browsersList: string[] = browserslist(queries, options);
+
+
+  return browsersList.some( (entry: string) => {
+    const [browserToSupport, versionToSupport] = parseBrowserVersionTuple(entry);
+
+    const minVersion = supportTable.get(browserToSupport);
+
+    if (typeof minVersion === 'undefined'
+      || cmpVersions(minVersion, versionToSupport) === 1) {
+      return true;
+    }
+
+    return false;
+  });
+}
+
+export function cmpVersions(version1: string, version2: string) {
+  const version1Tupel = toVersionTupel(version1);
+  const version2Tupel = toVersionTupel(version2);
+
+  if (version1Tupel[0] < version2Tupel[0]) {
+    return -1;
+  } else if (version1Tupel[0] > version2Tupel[0]) {
+    return 1;
+  } else if (version1Tupel[1] < version2Tupel[1]) {
+    return -1;
+  } else if (version1Tupel[1] > version2Tupel[1]) {
+    return 1;
+  } else {
+    return 0;
+  }
+}
+
+function toVersionTupel(version: string): [number, number] {
+  const temp = version.split('.');
+  if (temp.length === 1) {
+    temp[1] = '0';
+  }
+
+  return [parseInt(temp[0]), parseInt(temp[1])];
+}
+
+function parseBrowserVersionTuple(raw: string): [string, string] {
+  const strArray = raw.split(' ');
+  if (strArray[1] === 'all') {
+    strArray[1] = String(Number.MAX_SAFE_INTEGER);
+  }
+
+  return [strArray[0], strArray[1].split('-')[0]];
+}
+
+function convertSupportTableToBrowserVersionTuples(supportTable: object): Map<string, string> {
+  // The supportTable bases upon data from the caniuse api.
+  // It uses the y property to indicate the version beginning
+  // with the feature in question is supported.
+  const browsers = Object.getOwnPropertyNames(supportTable) as string[];
+  const supportMap = new Map<string, string>();
+
+  browsers
+     // @ts-ignore
```

Comment:
Why is this needed?

Reply:
We don't have typings for the data structures provided by caniuse. Hence, here it's just an object (supportTable: object). However, `supportTable[b].y` leads to an implicit any issue.

Reply:
What be good to do some internal interfaces, for type checking.

Reply:
+1